lib/srv/usermgmt.go

```diff
@@ -429,52 +344,63 @@ func (u *HostUserManagement) ensureGroupsExist(groups ...string) error {
 	return trace.NewAggregate(errs...)
 }
 
+// A HostUser represents all of the fields pertaining to an existing user on a host, including their group membership.
+type HostUser struct {
+	Name   string
+	UID    string
+	GID    string
+	Home   string
+	Groups map[string]struct{}
+}
```

Comment:
I plan on leveraging this struct for more refactors in the future, but for now it's mostly just here to capture which groups are assigned to an existing user.